core/trino-main/src/main/java/io/trino/util/AutoCloseableCloser.java

```diff
@@ -26,12 +26,21 @@
         implements AutoCloseable
 {
     private final Deque<AutoCloseable> stack = new ArrayDeque<>(4);
+    private final boolean reverseOrder;
 
-    private AutoCloseableCloser() {}
+    private AutoCloseableCloser(boolean reverseOrder)
+    {
+        this.reverseOrder = reverseOrder;
+    }
 
     public static AutoCloseableCloser create()
     {
-        return new AutoCloseableCloser();
+        return new AutoCloseableCloser(false);
```

Comment:
One could argue that `true` should be the default behavior. Or even the only one.

Reply:
Or I think I'm misunderstanding what's going on here :)

Reply:
if one of maintainers will agree to change default behaviour the flag would not be required.

Reply:
I guess I'm just confused. The way it works currently, it's already the case that the clients in `HiveMinioDataLake` will be closed before the server (the clients are registered after the server, and it's a LIFO queue, so they will be closed first). So how reversing that there would help?